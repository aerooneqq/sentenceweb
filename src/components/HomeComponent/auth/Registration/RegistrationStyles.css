#signUpInputContainer{
    width: 80%;
    margin-left: 10%;
    background: #f0f0f0;
}

.regInput{
  width:90%;
  height: 40px;

  margin-top: -10px;

  border-style: solid;
  border-width: 1px;
  border-color: orange;
  border-radius: 5px;

  font-size: 21px;
  font-family: Arial;

  padding-inline-start: 5%;
  padding-inline-end: 5%;
}

.regInput:hover {
  border-color: orangered;
  outline: none;
}

.regInput:focus {
  border-color: orangered;
  outline: none;
}


.regContentRotationThreeD {
  animation: regContentRotation 1s ease forwards
}

@keyframes regContentRotation {
  from {
    visibility: hidden;
    transform: rotate3d(0, 1, 0, 180deg)
  }
  
  100%{
    transform: rotate3d(0, 1, 0, 360deg)
  }
}

#signUpBtn{
  width: 100%;
  height: 60px;
  background: linear-gradient(32deg, orange, orangered);

  border-style: none;
  border-radius: 5px;

  font-family: Arial;
  color: white;
  font-size: 22px;
}

#signUpBtn:hover {
    background: linear-gradient(32deg, #FD8A60, #FC5B20);
    outline: none;
}

#signUpBtn:focus{
    outline: none;
}

.registrationLoaderCont { 
  margin-top: 100px;
  
  display: flex;
  align-items: center;
  justify-content: center;
}