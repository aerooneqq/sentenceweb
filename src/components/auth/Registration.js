import React, {Component} from "react"

export default class Registration extends Component{
  constructor(props){
    super(props)
  }

  componentDidMount(){
    document.getElementById("signUpInputContainer").classList.toggle("regContentRotationThreeD")
  }

  render(){
    return(
      <div id='signUpInputContainer'>
        <div className="inputPropContainer">
          <p className="inputNameText">E-mail</p>
          <input type='text' id="inputEmail" className="regInput"/>
        </div>
        <div className="inputPropContainer">
          <p className="inputNameText">Password</p>
          <input type='password' id='inputPassword' className="regInput"/>
        </div>
        <div className="inputPropContainer">
          <p className="inputNameText">Repeat password</p>
          <input type='password' id='inputPassword' className="regInput"/>
        </div>
        <div id="btnContainer">
          <button id="signUpBtn">Sign up</button>
          <p>By clicking the sign up btn u agree to freely give me all things u have</p>
        </div>
      </div>
    )
  }
}
