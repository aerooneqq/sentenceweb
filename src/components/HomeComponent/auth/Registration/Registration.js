import React, {Component} from "react"

//Styles
import "./RegistrationStyles.css"

//Components
import Loader from "../../../loader/Loader";
import UserService from "../../../../services/UserServices/UserService";

//App messages
import {alertAppMessage} from "../../../ApplicationMessage/ApplicationMessageManager";


export default class Registration extends Component{
  constructor(props){
    super(props);

    this.state = { 
      isLoading: false,
      email: "",
      password: "",
      repeatedPassword: ""
    }

    this.handleRegistrationClick = this.handleRegistrationClick.bind(this);
    this.handleEmailChange = this.handleEmailChange.bind(this);
    this.handlePasswordChange = this.handlePasswordChange.bind(this);
    this.handleRepeatPasswordChange = this.handleRepeatPasswordChange.bind(this);
  }

  handleRegistrationClick() { 
    let userService = new UserService(localStorage.getItem("token"));

    let {email, password, repeatedPassword} = this.state;

    if (password !== repeatedPassword) { 
      alertAppMessage("The password nust be the same", "error");
    }
    else { 
      this.setState({ 
        isLoading: true
      });

      userService.createNewUser(email, password).then(() => {
        alertAppMessage("The account was successfully created!", "success");
      }).catch(error => { 
        alert(error);
        if (error.response) { 
          for (let errorMsg of error.response.data) { 
            alertAppMessage(errorMsg, "error");
          }
        }
      }).then(() => { 
        this.setState({ 
          isLoading: false
        });
      });
    }
  }

  handleEmailChange({target: { value }}) { 
    this.setState({ 
      email: value
    });
  }

  handlePasswordChange({target: { value }}) { 
    this.setState({ 
      password: value
    });
  }

  handleRepeatPasswordChange({target: { value }}) { 
    this.setState({ 
      repeatedPassword: value
    });
  }

  render(){
    if (this.state.isLoading === true) { 
      return ( 
        <div className = "registrationLoaderCont">
          <Loader />
        </div>
      )
    }

    return (
      <div id='signUpInputContainer' className ="regContentRotationThreeD">
        <div className="inputPropContainer">
          <p className="inputNameText">E-mail</p>
          <input type='text' id="inputEmail" className="regInput" onChange = {this.handleEmailChange}/>
        </div>
        <div className="inputPropContainer">
          <p className="inputNameText">Password</p>
          <input type='password' id='inputPassword' className="regInput" onChange = {this.handlePasswordChange}/>
        </div>
        <div className="inputPropContainer">
          <p className="inputNameText">Repeat password</p>
          <input type='password' id='inputPassword' className="regInput" onChange = {this.handleRepeatPasswordChange}/>
        </div>
        <div id="btnContainer">
          <button id="signUpBtn" onClick = {this.handleRegistrationClick}>  
            Sign up
          </button> 
          <p>By clicking the sign up btn u agree to freely give me all things u have</p>
        </div>
      </div>
    )
  }
}
