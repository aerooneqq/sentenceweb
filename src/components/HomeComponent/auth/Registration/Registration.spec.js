import React from "react";
import { shallow } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import Registration from './Registration';

describe("Registration component", () => { 
    it("should render correctly", () => { 
        const registration = shallow(<Registration />)

        expect(shallowToJson(registration)).toMatchSnapshot();
    });

    it("should have right text", () => { 
        const registration = shallow(<Registration />);

        expect(registration.find("p.inputNameText").at(0).text()).toBe("E-mail");
        expect(registration.find("p.inputNameText").at(1).text()).toBe("Password");
        expect(registration.find("p.inputNameText").at(2).text()).toBe("Repeat password");
    });

    it("should have proper initial state", () => { 
        const registration = shallow(<Registration />);

        const rightState = { 
            isLoading: false,
            email: "",
            password: "",
            repeatedPassword: ""
        };

        for (let key in rightState) { 
            expect(rightState[key]).toBe(registration.state(key));
        }
    });

    it("should show loader if state is isLoading", () => { 
        const registration = shallow(<Registration />);

        registration.setState({ 
            isLoading: true
        });

        expect(registration.find("div").at(0).hasClass("registrationLoaderCont")).toBe(true);
    });
});
