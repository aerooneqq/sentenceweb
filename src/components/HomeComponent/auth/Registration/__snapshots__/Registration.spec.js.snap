// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component should render correctly 1`] = `
<div
  className="regContentRotationThreeD"
  id="signUpInputContainer"
>
  <div
    className="inputPropContainer"
  >
    <p
      className="inputNameText"
    >
      E-mail
    </p>
    <input
      className="regInput"
      id="inputEmail"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="inputPropContainer"
  >
    <p
      className="inputNameText"
    >
      Password
    </p>
    <input
      className="regInput"
      id="inputPassword"
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    className="inputPropContainer"
  >
    <p
      className="inputNameText"
    >
      Repeat password
    </p>
    <input
      className="regInput"
      id="inputPassword"
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    id="btnContainer"
  >
    <button
      id="signUpBtn"
      onClick={[Function]}
    >
      Sign up
    </button>
    <p>
      By clicking the sign up btn u agree to freely give me all things u have
    </p>
  </div>
</div>
`;
